File: Searching/Binary_Search/BinarySearch.py
#For binary search the values have to be sorted
#Time complexity O(log n)

index = -1                              #initializing index variable with one
def bsearch(lst,n):
    l = 0
    u = len(lst)-1
    while(l<=u):
        mid = (l+u)//2
        if lst[mid] == n:
            globals()['index'] = mid
            return True                 #returns True if mid value is equal to element of search
        else:
            if lst[mid]<n:
                l = mid+1
            else:
                u = mid - 1

lst = [2,6,7,10,154,298]
n = 154

if bsearch(lst,n):
    print("Element found at position",index+1)
else:
    print("Element not found")

File: Searching/Binary_Search/BinarySearch.c
#include<stdio.h>
int binarySearch(int arr[], int size, int element){
    int low, mid, high;
    low = 0;
    high = size-1;
    // Keep searching until low <= high
    while(low<=high){
        mid = (low + high)/2;
        if(arr[mid] == element){
            //Return the index of element if it is found
            return mid;
        }
        if(arr[mid]<element){
            low = mid+1;
        }
        else{
            high = mid -1;
        }
    } 
    //If element not fiund Return -1
    return -1;
    
}
int main(){
    // Note: Sorted array is Required for binary search
    int size; //Size of the arry
    scanf("%d",&size);
    int arr[size] ;
    for (int i = 0; i < size; i++)
    {
        scanf("%d", &arr[i]);
    }
    int target; //Element which we want to find
    scanf("%d",&target);
    int ans = binarySearch(arr, size, target); //Will return the index of target element if found
    printf("%d",ans);
    return 0;
}